src/bootsampling.jl

```diff
@@ -114,30 +114,30 @@ maximumEntropySampling(100, MaximumEntropyCache())
 NOTE: Implementation based off [pymeboot](https://github.com/kirajcg/pymeboot) as the original
 [R package](https://cran.r-project.org/web/packages/meboot/index.html) is GPL licensed.
 """
-struct MaximumEntropySampling <: BootstrapSampling
+struct MaximumEntropySampling{C<:MaximumEntropyCache} <: BootstrapSampling
     nrun::Int
-    cache::MaximumEntropyCache
+    cache::C
```

Comment:
This doesn't need to parameterized because `MaximumEntropyCache` is a concrete type (e.g., not like `AbstractFloat`).

Reply:
No, since the type parameter is not declared it's ambiguous. Like the example of `Point` in the [documentation](https://docs.julialang.org/en/v1/manual/types/index.html#Parametric-Composite-Types-1), the type object `MaximumEntropyCache` contains all instances `MaximumEntropyCache{Float64}`,`MaximumEntropyCache{Integer}`,etc. as subtypes.

Reply:
Oh, yeah. FWIW, I'm not sure how much this'll get us because the cache contains a bunch of container objects, but I guess it doesn't hurt. I'd probably do this though.

```
struct MaximumEntropySampling{T<:Real}
    nrun::Int
    cache::MaximumEntropyCache{T}
end
```